src/main/java/org/opensrp/web/rest/ClientFormResource.java

```diff
@@ -0,0 +1,157 @@
+        if (clientForm == null) {
+            return new ResponseEntity<>((String) null, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        completeClientForm = new ClientFormService.CompleteClientForm(clientForm, clientFormMetadata);
+
+        return new ResponseEntity<>(gson.toJson(completeClientForm), HttpStatus.OK);
+    }
+    
+    //@RequestMapping(consumes = {MediaType.MULTIPART_FORM_DATA_VALUE}, method = RequestMethod.POST)
+    @RequestMapping(headers = { "Accept=multipart/form-data" }, method = RequestMethod.POST)
+    @ResponseBody
+    private ResponseEntity<String> addClientForm(@RequestParam("form_version") String formVersion,
+                                                 @RequestParam("form_identifier") String formIdentifier,
+                                                 @RequestParam("form_name") String formName,
+                                                 @RequestParam("form") MultipartFile jsonFile,
+                                                 @RequestParam(required = false) String module) {
+        if (TextUtils.isEmpty(formVersion) || TextUtils.isEmpty(formIdentifier) || TextUtils.isEmpty(formName) || jsonFile.isEmpty()) {
+            return new ResponseEntity<>("Required params is empty", HttpStatus.BAD_REQUEST);
+        }
+
+        // TOOD: Check the string is a valid version
+        DefaultArtifactVersion formVersionV = new DefaultArtifactVersion(formVersion);
+        if (!jsonFile.getOriginalFilename().contains(".json")) {
+            return new ResponseEntity<>("The form is not a JSON file", HttpStatus.BAD_REQUEST);
+        }
+
+        if (jsonFile.isEmpty()) {
+            return new ResponseEntity<>("File is empty", HttpStatus.BAD_REQUEST);
+        }
+
+        ClientForm clientForm = new ClientForm();
+
+        byte[] bytes;
+        try {
+            bytes = jsonFile.getBytes();
+        } catch (IOException e) {
+            logger.error("Error occurred trying to read uploaded json file", e);
+            return new ResponseEntity<>("Invalid file", HttpStatus.BAD_REQUEST);
+        }
+
+        String jsonString = new String(bytes);
+        logger.info(jsonString);
+        clientForm.setJson(jsonString);
+        clientForm.setCreatedAt(new Date());
+
+        ClientFormMetadata clientFormMetadata = new ClientFormMetadata();
+        clientFormMetadata.setVersion(formVersion);
+        clientFormMetadata.setIdentifier(formIdentifier);
+        clientFormMetadata.setLabel(formName);
+        clientFormMetadata.setCreatedAt(new Date());
+        clientFormMetadata.setModule(module);
+
+        ClientFormService.CompleteClientForm completeClientForm = clientFormService.addClientForm(clientForm, clientFormMetadata);
+
+        if (completeClientForm == null) {
+            return new ResponseEntity<>("Unknown error. Kindly confirm that the form does not already exist on the server", HttpStatus.BAD_REQUEST);
```

Comment:
If this is an unknown error the 500 series code should be used. 

Make sure that `clientFormService.addClientForm` can throw specific exceptions that can be used here to determine what failed. 

If its because of wrong inputs then return  `HttpStatus.BAD_REQUEST` else `HttpStatus.INTERNAL_SERVER_ERROR`